### PR TITLE
[lambda][output] request_helper changes

### PR DESCRIPTION
**Pull request by armtash**
to @airbnb/streamalert-maintainers

size: small

* Added request_helper optional arg "headers" to pass http headers.
* Added request_helper optional arg "verify" to allow output to ignore
* server certificate checking. Default is verify=True
* Removed request_helper required arg "endpoint" as it's no longer needed.
* request_helper returns its urllib2 response to allow for outputs to
* chain http requests and better validate successful requests.

### CODE REVIEW

File: stream_alert_output/main.py [hunk 1]
```diff
@@ -120,7 +121,10 @@ def run(self, alerts):
                 output_func = getattr(self, '_{}'.format(output), None)
                 if output_func:
                     self._setup_output_creds(output)
-                    output_func(rule_name, formatted_alerts)
+                    try:
```
Comment (jacknagz) [inline stream_alert_output/main.py]: What is the error you're trying to catch here? I'd prefer a strategy where we catch specific errors in each respective `output`.

Comment (yeller15) [inline stream_alert_output/main.py]: I intended for it to be a general catchall.  An OutputRequestFailure exception is raised when a failure occurs in request_handler, which wasn't caught by anything.  I can modify this block to only catch OutputRequestFailure, or I can revert it to how it was before.

Comment (jacknagz) [inline stream_alert_output/main.py]: let's keep it how it was initially, otherwise the error raised would never actually fire

**Comment by jacknagz** [comment]
please run `pylint` on this commit, I noticed a couple offenses

**Comment by jacknagz** [comment]
@armtash the changes in here are also present in #66, perhaps we can just close this one out as a duplicate?

**Comment by armtash** [comment]
Sure

**Pull request closed without merging**